tests/lib/rule-tester/rule-tester.js

```diff
@@ -1906,6 +1907,24 @@ describe("RuleTester", () => {
         }, "Hydrated message \"Avoid using variables named 'notFoo'.\" does not match \"Avoid using variables named 'foo'.\"");
     });
 
+    it("should throw if the message has missing placeholders when data is not specified", () => {
+        assert.throws(() => {
+            ruleTester.run("foo", require("../../fixtures/testers/rule-tester/messageId").withMissingData, {
+                valid: [],
+                invalid: [{ code: "foo", errors: [{ messageId: "avoidFoo" }] }]
+            });
+        }, "The reported message has missing placeholders: name. Please provide them via the 'data' property in the context.report call");
```

Comment:
[Feature] Maybe I'm being too nitpicky here, but I think it'd be good to adjust the grammar to handle single vs. plural cases.

```suggestion
        }, "The reported message has a missing placeholder: name. Please provide it via the 'data' property in the context.report call");
```